go/vt/vtgate/planbuilder/route_planning.go

```diff
@@ -678,8 +678,14 @@ func canMergePlans(ctx planningContext, a, b *route) bool {
 		return false
 	}
 	switch a.eroute.Opcode {
-	case engine.SelectUnsharded, engine.SelectDBA, engine.SelectReference:
+	case engine.SelectUnsharded, engine.SelectReference:
 		return a.eroute.Opcode == b.eroute.Opcode
+	case engine.SelectDBA:
+		return b.eroute.Opcode == engine.SelectDBA &&
+			len(a.eroute.SysTableTableSchema) == 0 &&
+			len(a.eroute.SysTableTableName) == 0 &&
+			len(b.eroute.SysTableTableSchema) == 0 &&
+			len(b.eroute.SysTableTableName) == 0
```

Comment:
this merge logic could be improved by checking the values and if they point to the same name and schema, then it can be merged.